examples_tests/61.BlitFilterTest/main.cpp

```diff
@@ -0,0 +1,1561 @@
+	imageParams.type = imageType;
+	imageParams.format = format;
+	imageParams.extent = { dims[0], dims[1], dims[2] };
+	imageParams.mipLevels = 1u;
+	imageParams.arrayLayers = 1u;
+	imageParams.samples = asset::ICPUImage::ESCF_1_BIT;
+
+	auto imageRegions = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<asset::IImage::SBufferCopy>>(1ull);
+	auto& region = (*imageRegions)[0];
+	region.bufferImageHeight = 0u;
+	region.bufferOffset = 0ull;
+	region.bufferRowLength = dims[0];
+	region.imageExtent = { dims[0], dims[1], dims[2] };
+	region.imageOffset = { 0u, 0u, 0u };
+	region.imageSubresource.baseArrayLayer = 0u;
+	region.imageSubresource.layerCount = 1u;
+	region.imageSubresource.mipLevel = 0;
+
+	size_t bufferSize = asset::getTexelOrBlockBytesize(imageParams.format) * static_cast<size_t>(region.imageExtent.width) * region.imageExtent.height * region.imageExtent.depth;
+	auto imageBuffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(bufferSize);
+	core::smart_refctd_ptr<ICPUImage> image = ICPUImage::create(std::move(imageParams));
+	image->setBufferAndRegions(core::smart_refctd_ptr(imageBuffer), imageRegions);
+
+	if (fillWithTestData)
+	{
+		double pixelValueUpperBound = 20.0;
+		if (asset::isNormalizedFormat(format))
+			pixelValueUpperBound = 1.00000000001;
+
+		std::uniform_real_distribution<double> dist(0.0, pixelValueUpperBound);
+		std::mt19937 prng;
+
+		uint8_t* bytePtr = reinterpret_cast<uint8_t*>(image->getBuffer()->getPointer());
+		for (uint64_t k = 0u; k < dims[2]; ++k)
+		{
+			for (uint64_t j = 0u; j < dims[1]; ++j)
+			{
+				for (uint64_t i = 0; i < dims[0]; ++i)
+				{
+					double decodedPixel[4] = { 0 };
+					for (uint32_t ch = 0u; ch < asset::getFormatChannelCount(format); ++ch)
+						decodedPixel[ch] = dist(prng);
+
+					const uint64_t pixelIndex = (k * dims[1] * dims[0]) + (j * dims[0]) + i;
+					asset::encodePixelsRuntime(format, bytePtr + pixelIndex * asset::getTexelOrBlockBytesize(format), decodedPixel);
+				}
+			}
+		}
+	}
+
+	return image;
+}
+
+static inline asset::IImageView<asset::ICPUImage>::E_TYPE getImageViewTypeFromImageType_CPU(const asset::IImage::E_TYPE type)
+{
+	switch (type)
+	{
+	case asset::IImage::ET_1D:
+		return asset::ICPUImageView::ET_1D;
+	case asset::IImage::ET_2D:
+		return asset::ICPUImageView::ET_2D;
+	case asset::IImage::ET_3D:
+		return asset::ICPUImageView::ET_3D;
+	default:
+		__debugbreak();
+		return static_cast<asset::IImageView<asset::ICPUImage>::E_TYPE>(0u);
+	}
+}
+
+static inline video::IGPUImageView::E_TYPE getImageViewTypeFromImageType_GPU(const video::IGPUImage::E_TYPE type)
+{
+	switch (type)
+	{
+	case video::IGPUImage::ET_1D:
+		return video::IGPUImageView::ET_1D;
+	case video::IGPUImage::ET_2D:
+		return video::IGPUImageView::ET_2D;
+	case video::IGPUImage::ET_3D:
+		return video::IGPUImageView::ET_3D;
+	default:
+		__debugbreak();
+		return static_cast<video::IGPUImageView::E_TYPE>(0u);
+	}
+}
+
+class CBlitFilter
```

Comment:
I guess it would go into a `nbl/video/utilities/CComputeBlit` and be called `CComputeBlit`

Reply:
Should this be derived from `IReferenceCounted`, like other utilities?

Reply:
yes